### ISSUE TITLE
[CMake] VTK 9.0 deprecation warnings

### ISSUE THREAD

**Issue by SunBlack** [PointCloudLibrary/pcl#4046]
I tested VTK 9.0 on my Windows machine and I'm getting now:

```
CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkChartsCore' component is 'ChartsCore'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkCommonCore' component is 'CommonCore'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkCommonDataModel' component is 'CommonDataModel'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkCommonExecutionModel' component is
  'CommonExecutionModel'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkFiltersCore' component is 'FiltersCore'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkFiltersExtraction' component is
  'FiltersExtraction'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkFiltersModeling' component is 'FiltersModeling'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkImagingCore' component is 'ImagingCore'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkImagingSources' component is 'ImagingSources'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkInteractionStyle' component is 'InteractionStyle'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkInteractionWidgets' component is
  'InteractionWidgets'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkIOCore' component is 'IOCore'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkIOGeometry' component is 'IOGeometry'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkIOImage' component is 'IOImage'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkIOLegacy' component is 'IOLegacy'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkIOPLY' component is 'IOPLY'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkRenderingAnnotation' component is
  'RenderingAnnotation'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkRenderingLOD' component is 'RenderingLOD'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkViewsContext2D' component is 'ViewsContext2D'
Call Stack (most recent call first):
  CMakeLists.txt:390 (find_package)


CMake Warning (dev) at D:/vtk_install/lib/cmake/vtk-9.0/VTK-targets.cmake:28 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "" will no longer be dereferenced when the policy is
  set to NEW.  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:126 (include)
  CMakeLists.txt:390 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at D:/vtk_install/lib/cmake/vtk-9.0/VTK-targets.cmake:36 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "" will no longer be dereferenced when the policy is
  set to NEW.  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:126 (include)
  CMakeLists.txt:390 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkChartsCore' component is 'ChartsCore'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkCommonCore' component is 'CommonCore'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkCommonDataModel' component is 'CommonDataModel'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkCommonExecutionModel' component is
  'CommonExecutionModel'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkFiltersCore' component is 'FiltersCore'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkFiltersExtraction' component is
  'FiltersExtraction'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkFiltersModeling' component is 'FiltersModeling'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkImagingCore' component is 'ImagingCore'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkImagingSources' component is 'ImagingSources'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkInteractionStyle' component is 'InteractionStyle'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkInteractionWidgets' component is
  'InteractionWidgets'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkIOCore' component is 'IOCore'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkIOGeometry' component is 'IOGeometry'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkIOImage' component is 'IOImage'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkIOLegacy' component is 'IOLegacy'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkIOPLY' component is 'IOPLY'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkRenderingAnnotation' component is
  'RenderingAnnotation'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkRenderingLOD' component is 'RenderingLOD'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkViewsContext2D' component is 'ViewsContext2D'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake:64 (message):
  The new name for the 'vtkRenderingContextOpenGL' component is
  'RenderingContextOpenGL'
Call Stack (most recent call first):
  CMakeLists.txt:413 (find_package)


CMake Warning at CMakeLists.txt:413 (find_package):
  Found package configuration file:

    D:/vtk_install/lib/cmake/vtk-9.0/vtk-config.cmake

  but it set VTK_FOUND to FALSE so package "VTK" is considered to be NOT
  FOUND.  Reason given by package:

  Could not find the VTK package with the following required components:
  RenderingContextOpenGL.



VTK_MAJOR_VERSION 9, rendering backend: OpenGL
CMake Deprecation Warning at D:/vtk_install/lib/cmake/vtk-9.0/vtk-use-file-deprecated.cmake:1 (message):
  The `VTK_USE_FILE` is no longer used starting with 8.90.
Call Stack (most recent call first):
  CMakeLists.txt:418 (include)
```

**Comment by larshg** [comment]
There is already some progress on this, see https://github.com/PointCloudLibrary/pcl/pull/3637.
Is 9.0 release officially now?

**Comment by SunBlack** [comment]
Yeah, the git tag for 9.0 seems to be 4 days old.

**Comment by kunaltyagi** [comment]
🙈 We're not ready for even VTK 8

**Comment by SunBlack** [comment]
Well I didn't had trouble with MSVC & VTK 8 until now - I was compiling the PCL since a year or so with VTK 8.2 (you just have to enable legacy modules during compiling VTK).

**Comment by kunaltyagi** [comment]
BTW, the warnings seem like CMake targets not the code. Am I right in this?

**Comment by SunBlack** [comment]
Yes this are only warnings. The only issue: This code doesn't work anymore:
https://github.com/PointCloudLibrary/pcl/blob/b6962198a130c0349bea5e5f02415d77ca62cbf2/CMakeLists.txt#L404-L411

I think we should create a `pcl_find_vtk.cmake` and in this sth. like this:
````
find_package(VTK 9.0 COMPONENTS ${PCL_VTK_COMPONENTS})
if (NOT VTK_FOUND)
  // add vtk before each component
  find_package(VTK 6.2 COMPONENTS ${PCL_VTK_COMPONENTS_WITH_VTK_PREFIX})
endif()
````

**Comment by kunaltyagi** [comment]
I'll leave this to CMake experts now

**Comment by SunBlack** [comment]
> 🙈 We're not ready for even VTK 8

Just to fix this: We are not compatible to VTK 8.2 or newer. 8.1 or older are supported. Reason is this short [commit](https://gitlab.kitware.com/vtk/vtk/-/commit/8ea1565643abe793b44d265fbccb10df7a3ad22a) ;-)

**Comment by larshg** [comment]
Closed by #4262.
